crates/next-core/src/app_page_loader_tree.rs

```diff
@@ -250,7 +250,7 @@ impl AppPageLoaderTreeBuilder {
         let metadata_route = &*get_metadata_route_name((*item).into()).await?;
         writeln!(
             self.loader_tree_code,
-            "{s}  url: fillMetadataSegment({}, props.params, {}) + \
+            "{s}  url: fillMetadataSegment({}, await props.params, {}) + \
```

Comment:
Seems safe since we mark the function as `async` on L244. Would still be nice to get confirmation from @huozhi that we currently test this codepath.

Reply:
The metadata tests are currently not run with dynamicIO yet as there're still some untackled issues yet. But ideally we want to do that later.

Reply:
Is this codepath only hit with dynamic I/O? I'm not too concerned with fixing the warning since that's fairly obvious. What's not obvious is that `await` is actually in an async function so I want to make sure we have some smoke test. 

Or you feel confident to merge this without a test.

Reply:
It's fine to merge now as the webpack loader side was also updated before with the same logic. we can follow up the testing later